Libs/MRML/Core/vtkMRMLDisplayableNode.cxx

```diff
@@ -292,6 +292,7 @@ void vtkMRMLDisplayableNode::SetDisplayVisibility(int visible)
       displayNode->SetVisibility(visible);
       }
     }
+  this->Modified();
```

Comment:
You are right, that was quite bad ( I should have understood when I saw recompiling many things)! I moved the Modified event in the slot onROICropDisplayCheckBoxToggled, now should be fine.
Thanks!